File: solutions/172.Factorial_Trailing_Zeroes/AC_math_logn.java
/*
*  Author:      illuz <iilluzen[at]gmail.com>
*  File:        AC_math_logn.java
*  Create Date: 2015-02-05 11:44:55
*  Descripton:   
*/

import java.util.Scanner;

public class Solution {

    public int trailingZeroes(int n) {
        int ans = 0;
        while (n != 0) {
            ans += n / 5;
            n /= 5;
        }
        return ans;
    }

    // debug
    public static void main(String[] args) {
        Scanner cin = new Scanner(System.in);
        Solution s = new Solution();
        System.out.println("no case");
    }

}

File: solutions/172.Factorial_Trailing_Zeroes/AC_math_logn.py
# Author:      illuz <iilluzen[at]gmail.com>
# File:        AC_math_logn.py
# Create Date: 2015-02-05 11:45:02
# Descripton:   

class Solution:
    # @return an integer
    def trailingZeroes(self, n):
        ans = 0
        while n:
            ans += n // 5
            n //= 5
        return ans

# debug
s = Solution()
print s.trailingZeroes(5)

